### PR TITLE
Filter closed trusts

### PR DESCRIPTION
**Pull request by nwarms**
## Changes
Excludes everything but OPEN trusts at a db level
Context's model builders are now split into statically called partials based on entities
Adjusted Cypress test insertScript to be 'slightly' more manageable (To be scrapped at a later date).

[Ticket](https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_sprints/taskboard/Find%20Information%20about%20Academies%20and%20Trusts/Academies-and-Free-Schools-SIP/Find%20information%20about%20academies%20and%20trusts/Sprint%202?workitem=171803)
## Checklist

- [x] Add release notes to CHANGELOG.md

### CODE REVIEW

File: DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/DataSourceProvider.cs [hunk 1]
```diff
@@ -10,8 +11,9 @@ public class DataSourceProvider : IDataSourceProvider
     private readonly ILogger<DataSourceProvider> _logger;
 
     [ExcludeFromCodeCoverage]
-    public DataSourceProvider(AcademiesDbContext academiesDbContext, ILogger<DataSourceProvider> logger) : this(
-        (IAcademiesDbContext)academiesDbContext, logger)
+    public DataSourceProvider(AcademiesDbContext academiesDbContext, ILogger<DataSourceProvider> logger) :
+        this(
+            (IAcademiesDbContext)academiesDbContext, logger)
```
Comment (dynamictulip) [inline DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/DataSourceProvider.cs:16]: Can we merge line 15 and 16?

File: tests/DfE.FindInformationAcademiesTrusts.Data.AcademiesDb.Faker/Fakers/GiasGroupFaker.cs [hunk 2]
```diff
@@ -30,6 +33,7 @@ public GiasGroup Generate(TrustToGenerate trustToGenerate, int uid)
         fakeGiasGroup.GroupName = trustToGenerate.Name;
         fakeGiasGroup.GroupType = trustToGenerate.TrustType;
         fakeGiasGroup.GroupUid = uid.ToString();
+        fakeGiasGroup.GroupStatusCode = "OPEN";
```
Comment (dynamictulip) [inline tests/DfE.FindInformationAcademiesTrusts.Data.AcademiesDb.Faker/Fakers/GiasGroupFaker.cs:36]: This isn't needed as it's set in the rules

File: tests/DfE.FindInformationAcademiesTrusts.Data.AcademiesDb.Faker/AcademiesDbData.cs [hunk 3]
```diff
@@ -26,7 +27,8 @@ public class AcademiesDbData
     public IAcademiesDbContext AsAcademiesDbContext()
     {
         return new AcademiesDbDataContext(GiasEstablishments, GiasGovernances, GiasGroupLinks, GiasGroups, MstrTrusts,
-            CdmAccounts, MisEstablishments, MisFurtherEducationEstablishments, CdmSystemusers, MstrTrustGovernances, ApplicationEvents,
+            CdmAccounts, MisEstablishments, MisFurtherEducationEstablishments, CdmSystemusers, MstrTrustGovernances,
+            ApplicationEvents,
             ApplicationSettings);
```
Comment (dynamictulip) [inline tests/DfE.FindInformationAcademiesTrusts.Data.AcademiesDb.Faker/AcademiesDbData.cs:32]: Can we merge line 31 and 32?

**Review by dynamictulip — COMMENTED** [review]
Looks good, loving the academies db context split out.
Some minor comments, also can the PR description be updated?

File: tests/DfE.FindInformationAcademiesTrusts.Data.AcademiesDb.Faker/Fakers/GiasGroupFaker.cs (hunk 2 above)
Comment (dneed-nimble) [reply tests/DfE.FindInformationAcademiesTrusts.Data.AcademiesDb.Faker/Fakers/GiasGroupFaker.cs:36]: Removed

File: DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/DataSourceProvider.cs (hunk 1 above)
Comment (dneed-nimble) [reply DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/DataSourceProvider.cs:16]: Merged

**Review by dynamictulip — APPROVED** [review]

**Pull request merged**